actionpack/lib/action_dispatch/system_testing/server.rb

```diff
@@ -10,31 +8,17 @@ class << self
       self.silence_puma = false
 
       def run
-        register
         setup
       end
 
       private
-        def register
-          Capybara.register_server :rails_puma do |app, port, host|
```

Comment:
We can't remove this server. If someone is already using it directly they expect it to continue working. Is the Rails puma handler a problem that we need to remove it? Why?

Reply:
@rafaelfranca The issue is that  `Rack::Handler::Puma.run` traps interrupt signals which prevents the tests from ending when ctrl-c is pressed.

Reply:
The justification for removing the puma handler is in the relevant commit that I backported from master. Your point about backwards compatibility is legit though, so I'll just close this PR and monkeypatch! Thanks.